shotover-proxy/src/transforms/cassandra/sink_cluster/topology.rs

```diff
@@ -103,68 +70,281 @@ async fn topology_task_process(
+                            data_center_index.context("data_center missing from columns")?,
+                            tokens_index.context("tokens missing from columns")?,
+                            broadcast_port_index,
+                        )
                     };
 
-                    Ok(CassandraNode {
-                        address: SocketAddr::new(ip, 9042),
-                        rack,
-                        _tokens: tokens,
-                        outbound: None,
+                    rows.iter_mut()
+                        .filter(|row| {
+                            if let Some(MessageValue::Varchar(data_center)) =
+                                row.get(data_center_index)
+                            {
+                                data_center == config_data_center
+                            } else {
+                                false
+                            }
+                        })
+                        .map(|row| {
+                            let tokens =
+                                if let Some(MessageValue::List(list)) = row.get(tokens_index) {
+                                    list.iter()
+                                        .map::<Result<String>, _>(|x| match x {
+                                            MessageValue::Varchar(a) => Ok(a.clone()),
+                                            _ => Err(anyhow!("tokens value not a varchar")),
+                                        })
+                                        .collect::<Result<Vec<String>>>()?
+                                } else {
+                                    return Err(anyhow!("tokens not a list"));
+                                };
+
+                            let rack =
+                                if let Some(MessageValue::Varchar(value)) = row.get(rack_index) {
+                                    value.clone()
+                                } else {
+                                    return Err(anyhow!("rack not a varchar"));
+                                };
+
+                            let ip = if let Some(MessageValue::Inet(value)) =
+                                row.get(broadcast_address_index)
+                            {
+                                *value
+                            } else {
+                                return Err(anyhow!("address not an inet"));
+                            };
+
+                            let port = if let Some(broadcast_port_index) = broadcast_port_index {
+                                if let Some(MessageValue::Integer(value, _)) =
+                                    row.get(broadcast_port_index)
+                                {
+                                    *value
+                                } else {
+                                    return Err(anyhow!("port not an integer"));
+                                }
+                            } else {
+                                9042
```

Comment:
can we add a comment just above the 9042, like: "cassandra 3.x does not have the broadcast_port field so we just assume the default port.